xunit-autolabeler-v2/ast_parser/core/cli_yaml.py

```diff
@@ -0,0 +1,302 @@
+
+    is_valid = len(errors) == 0
+    return (is_valid, errors)
+
+
+def _validate_region_tags(
+    yaml_paths: List[str],
+    grep_tags: List[str],
+    source_tags: List[str],
+) -> Tuple[bool, List[str]]:
+    """Validate region-tag keys in a list of .drift-data.yml files
+
+    This method verifies that region tag keys within a .drift-data.yml
+    file are used *exactly* once in the source code.
+
+    Args:
+        yaml_paths: A list of .drift-data.yml filepaths to validate
+        grep_tags: A list of region tags (parsed *and* unparsed) that exist
+                   in the target directory
+        source_tags: A list of successfully-parsed region tags in the target
+                     directory
+
+    Returns:
+        A 2-tuple containing the following:
+         - A boolean including file validity (True if all files passed region
+           tag validation, False otherwise)
+         - A list of validation error messages (if any) that were raised
+    """
+
+    seen_region_tags = set()
+    output = []
+
+    for yaml_path in yaml_paths:
+        with open(yaml_path, 'r') as file:
+            yaml_contents = '\n'.join(file.readlines())
+            parsed_yaml = yaml.safe_load(yaml_contents)
+
+            for tag in parsed_yaml.keys():
+                yaml_entry = parsed_yaml[tag]
+                tag_should_be_in_source = not (
+                    'tested' in yaml_entry and yaml_entry['tested'] is False)
+
+                # Verify mentioned region tags are used in the
+                # source code (via parsing and/or grep results)
+                if tag not in grep_tags:
+                    output.append(
+                        f'Yaml file {yaml_path} contains region '
+                        f'tag not used in source files: {tag}')
+                    is_valid = False
+                elif tag_should_be_in_source and tag not in source_tags:
+                    output.append(
+                        f'Yaml file {yaml_path} contains '
+                        f'unparsed region tag: {tag}')
+                    output.append(
+                        '  Remove it, or label it with "tested: false".')
+                    is_valid = False
+                elif not tag_should_be_in_source and tag in source_tags:
+                    output.append(f'Parsed tag {tag} in file'
+                                  f'{yaml_path} marked untested!')
+                    is_valid = False
+
+                # Verify region tags are present at most once
+                if tag in seen_region_tags:
+                    output.append(f'Region tag {tag} is used multiple '
+                                  'times in .drift-data.yml files!')
+                    is_valid = False
+                else:
+                    seen_region_tags.add(tag)
+
+    return (is_valid, output)
+
+
+def validate_yaml_syntax(
+    root_dir: str,
+    grep_tags: List[str],
+    source_tags: List[str]
+) -> Tuple[bool, List[str]]:
```

Comment:
Looks like this public method is not being tested?

Reply:
It's being tested indirectly through calls to the CLI method in `cli.py`.

(It's a fairly straightforward method, so I'm not sure there's much value in testing it directly. That said, I can add a trivial test or two if you'd like.)

Reply:
Oh I see, I agree is trivial and has a single branch so that should be enough.